x-pack/plugins/security_solution/scripts/endpoint/common/fleet_server/fleet_server_services.ts

```diff
@@ -0,0 +1,673 @@
+
+    if (existingFleetServerIntegrationPolicy) {
+      log.verbose(
+        `Found existing Fleet Server Policy: ${JSON.stringify(
+          existingFleetServerIntegrationPolicy,
+          null,
+          2
+        )}`
+      );
+      log.info(
+        `Using existing Fleet Server agent policy id: ${existingFleetServerIntegrationPolicy.policy_id}`
+      );
+
+      return existingFleetServerIntegrationPolicy.policy_id;
+    }
+
+    log.info(`Creating new Fleet Server policy`);
+
+    const createdFleetServerPolicy: AgentPolicy = await kbnClient
+      .request<CreateAgentPolicyResponse>({
+        method: 'POST',
+        path: AGENT_POLICY_API_ROUTES.CREATE_PATTERN,
+        headers: { 'elastic-api-version': '2023-10-31' },
+        body: {
+          name: `Fleet Server policy (${Math.random().toString(32).substring(2)})`,
+          description: `Created by CLI Tool via: ${__filename}`,
+          namespace: 'default',
+          monitoring_enabled: ['logs', 'metrics'],
+          // This will ensure the Fleet Server integration policy
+          // is also created and added to the agent policy
+          has_fleet_server: true,
+        },
+      })
+      .then((response) => response.data.item)
+      .catch(catchAxiosErrorFormatAndThrow);
+
+    log.info(
+      `Agent Policy created: ${createdFleetServerPolicy.name} (${createdFleetServerPolicy.id})`
+    );
+    log.verbose(createdFleetServerPolicy);
+
+    return createdFleetServerPolicy.id;
+  });
+};
+
+interface StartFleetServerWithDockerOptions {
+  kbnClient: KbnClient;
+  logger: ToolingLog;
+  policyId: string;
+  serviceToken: string;
+  version?: string;
+  port?: number;
+}
+
+const startFleetServerWithDocker = async ({
+  kbnClient,
+  logger: log,
+  policyId,
+  serviceToken,
+  version,
+  port = 8220,
+}: StartFleetServerWithDockerOptions): Promise<StartedServer> => {
+  await verifyDockerInstalled(log);
+
+  let agentVersion = version || (await getAgentVersionMatchingCurrentStack(kbnClient));
+
+  log.info(`Starting a new fleet server using Docker (version: ${agentVersion})`);
+
+  const response: StartedServer = await log.indent(4, async () => {
+    const isServerless = await isServerlessKibanaFlavor(kbnClient);
+    const localhostRealIp = getLocalhostRealIp();
+    const fleetServerUrl = `https://${localhostRealIp}:${port}`;
+    const esURL = new URL(await getFleetElasticsearchOutputHost(kbnClient));
+    const containerName = `dev-fleet-server.${port}`;
+    const hostname = `dev-fleet-server.${port}.${Math.random().toString(32).substring(2, 6)}`;
+    let containerId = '';
+
+    if (isLocalhost(esURL.hostname)) {
+      esURL.hostname = localhostRealIp;
+    }
+
+    if (isServerless) {
+      log.info(`Kibana running in serverless mode.
+  - will install/run standalone Fleet Server
+  - version adjusted to [latest] from [${agentVersion}]`);
+
+      agentVersion = 'latest';
+      await maybeCreateDockerNetwork(log);
```

Comment:
Is there any case where we need to additionally create the network (in serverless mode)? I understand that network is being already created with dockerized ES.

Reply:
Not sure 🤷  . I think there are some additional discussions in @patrykkopycinski 's PR about creating additional networks for Agent (hosts).